lib/charms/smtp_integrator/v0/smtp.py

```diff
@@ -170,6 +170,7 @@ def to_relation_data(self) -> Dict[str, str]:
         if self.password:
             result["password"] = self.password
         if self.password_id:
+            result.pop("password", None)
```

Comment:
I'm wondering if it would help log this so that the user knows to set either one, not both